### ISSUE TITLE
`driftctl scan` crashed when used with many remote tfstate files

### ISSUE THREAD

**Issue by theintz** [cloudskiff/driftctl#469]
**Description**

Our TF setup is split up into multiple directories which each comprise a part of the overall cloud system all contained in the same AWS account. There are directories for global/, dev/, prod/, etc. Each directory has its own tfstate file which is stored on S3.

When I try to invoke `driftctl scan` passing all these tfstate files with the `--from tfstate+s3://[bucket]/[key.tfstate]` syntax, the tool can handle up to 4 of these parameters, but crashes when I try to use it with 5 or more.

The error message given is: "A runner routine paniced: value is marked, so must be unmarked first"

**Environment**
 - OS: MacOS 11.3
 - driftctl version: 0.7.1

**How to reproduce**

See above

**Possible Solution**

**Additional context**

**Comment by eliecharra** [comment]
@theintz Do you reproduce the issue when using only one `--from` flag with the fifth state ? I suspect an issue with the latest state.

**Comment by theintz** [comment]
You are right, the same error happens when I use just that one state file. What can I do to help you reproduce?

**Comment by eliecharra** [comment]
Which version of terraform and which version of the AWS provider you used for the problematic state ?

Maybe you can use driftctl with crash reporting flag https://docs.driftctl.com/0.7.1/usage/flags/crash-reporting, it will send us a crash report and maybe we can investigate further. But it looks like it's an issue within terraform.

**Comment by theintz** [comment]
Here is the output
```➜ tf version
Terraform v0.15.0
on darwin_amd64
+ provider registry.terraform.io/hashicorp/archive v2.1.0
+ provider registry.terraform.io/hashicorp/aws v3.35.0

➜ tf providers

Providers required by configuration:
.
├── provider[registry.terraform.io/hashicorp/archive]
└── provider[registry.terraform.io/hashicorp/aws] ~> 3.35.0

Providers required by state:

    provider[registry.terraform.io/hashicorp/archive]

    provider[registry.terraform.io/hashicorp/aws]
```

I will run driftctl with the crash flag enabled.

**Comment by eliecharra** [comment]
> Terraform v0.15.0

Maybe this is an issue with the latest terraform version,we need to check this on our side.

> I will run driftctl with the crash flag enabled.

Thanks, we'll stay you informed.

Anyway, thanks for your feedback !

**Comment by eliecharra** [comment]
We recieved your crash report thanks !

![image](https://user-images.githubusercontent.com/6154987/116977948-56cfef00-acc3-11eb-8a0b-864b509dc161.png)

**Comment by eliecharra** [comment]
@theintz It look like this is an issue within terraform.
I just build for you a driftctl version with an updated terraform version, can you run it on your state and let me know if it fixes your issue ?

[driftctl_darwin_amd64.zip](https://github.com/cloudskiff/driftctl/files/6421900/driftctl_darwin_amd64.zip)

If it solve your issue I will bump terraform version and the fix will be available in upcoming releases.

Thanks for your time 🙏🏻

**Comment by theintz** [comment]
Hi @eliecharra, I am unable to run the executable since it is not verified. Maybe you could push a proper update?

**Comment by eliecharra** [comment]
> Hi @eliecharra, I am unable to run the executable since it is not verified. Maybe you could push a proper update?

By verified what do you mean exactly ? This is a dev binay and thus it is not signed with our GPG key but it does not matter.

**Comment by theintz** [comment]
MacOS tells me that it's unverified and can't be run. I'm not comfortable with adding a security exception.

**Comment by eliecharra** [comment]
I don't get it, driftctl is not signed with an apple account even for public releases, so you have to add it as a security exception anyway. I'm afraid I can't help you further if you cannot confirm that this build fix your issue.

**Comment by sundowndev** [comment]
> MacOS tells me that it's unverified and can't be run. I'm not comfortable with adding a security exception.

May be it'd be easier for you to compile driftctl by yourself on a dedicated branch so you don't have to worry about security ? You just need to install Go and clone the git repository.

**Comment by sundowndev** [comment]
Hi @theintz , did you try to use the glob pattern feature introduced in v0.8 ? Instead of passing multiple instances of `--from` like so : `--from tfstate+s3://path/to/states/**/*.tfstate` ([documentation](https://docs.driftctl.com/0.9.0/usage/cmd/scan-usage#--from)). Let us know if you still have issues with your use case.
